Client/Helpers/FxALoginHelper.swift

```diff
@@ -54,15 +54,21 @@ class FxALoginHelper {
     // to changing of user settings and push notifications.
     func application(_ application: UIApplication, didLoadProfile profile: Profile) {
         self.profile = profile
+        self.account = profile.getAccount()
 
-        guard let account = profile.getAccount() else {
+        guard let account = self.account else {
             // There's no account, no further action.
             return loginDidFail()
         }
 
         // accountVerified is needed by delegates.
         accountVerified = account.actionNeeded != .needsVerification
 
+        // We should check if deviceRegistration has been performed, and 
+        // update the sync scratch pad (a proxy for our client record) accordingly.
+        // We do this here because this is effectively the upgrade path between 7 and 8.
+        updateSyncScratchpad()
+
         guard AppConstants.MOZ_FXA_PUSH else {
             return loginDidSucceed()
         }
```

Comment:
Does this also need an explicit check if fxaDeviceId has been set and uploaded? Or is that guaranteed when we have one the push registration?